syntax = "proto3";

package schema;

import "google/api/annotations.proto";

// Contains the name and content of a *.proto file
message ProtoFile {
    string file_name = 1;  // name of proto file
    string proto = 2;  // content of proto file
    bytes descriptor = 3;  // compiled descriptor for proto (zlib compressed)
}

// Proto files and compiled descriptors for this interface
message Schemas {

    // Proto files
    repeated ProtoFile protos = 1;

}

// Empty message
message NullMessage {}

// Schema services
service SchemaService {

    // Return active grpc schemas
    rpc GetSchema(NullMessage) returns (Schemas) {
        option (google.api.http) = {
            get: "/schema"
        };
    }

}
